### ISSUE TITLE
Junit 5 AllDisabled not detected if placed on parent class

### ISSUE THREAD

**Issue by smougenot** [togglz/togglz#303]
It's nice to have support for Junit 5 ... even if it's difficult to find out in the doc.

I''m facing an issue with annotation `@AllDisabled` (same with its conterpart `@AllEnabled`).

Working :
* add the annotation on the test method 
* add the annotation on the class containing the test method 

Not Working :
* add the annotation on any parent (abstract) class of the class containing the test method 

More details :
Let say we have an abstract class for tests called `AbstractParentTest`. It contains some logic and tooling shared with all the tests.
Adding a test class `MyTest` that inherit `AbstractParentTest`.
Now I'd like to disable/enable all the `Feature` for a particular test method.
The failing case is when `@AllDisabled` is placed on `AbstractParentTest` type.
- Features are not disabled
- we have `null` value for a parameter `TestFeatureManager` (on test related methods)

**Comment by chkal** [comment]
Hey @rweisleder, any thoughts about this?

**Comment by rweisleder** [comment]
Thanks for the feedback, @smougenot!

You're right, there is no real documentation at the moment. I've created https://github.com/togglz/togglz-site/issues/21 for this.

As you figured out, `@AllEnabled` and `@AllDisabled` only work when present on the concrete test class/method. Also, an annotation on method-level overrides an annotation on class-level. I'm not really happy with the current solution.

I've already thought about an enhanced evaluation of annotations. Something like
```java
@AllEnabled(MyFeatures.class)
abstract class ParentTest {

  @Test
  void testSomething(TestFeatureManager featureManager) {
    // ...
  }
}

@AllDisabled(OtherFeatures.class)
class ChildTest extends ParentTest {

  @Test
  @AllDisabled(MyFeatures.class)
  void testSomethingDifferent(TestFeatureManager featureManager) {
    // ...
  }
}
```

- `ChildTest.testSomething` would evaluate `@AllEnabled(MyFeatures.class)` and `@AllDisabled(OtherFeatures.class)` in that order.
- `ChildTest.testSomethingDifferent` would evaluate `@AllEnabled(MyFeatures.class)`, `@AllDisabled(OtherFeatures.class)` and `@AllDisabled(MyFeatures.class)` in that order.

**Comment by rweisleder** [comment]
Nevermind, my suggestion will not work because the `TestFeatureManager` expects a single enum class...

**Comment by chkal** [comment]
Fixed by #305

**Comment by smougenot** [comment]
Great, thank you for this fast answer
Great job!
